### DISCUSSION TITLE
Dynamic DNS with Unifi Dream Machine

### DISCUSSION THREAD

**Discussion by ConGp20** [TechnitiumSoftware/DnsServer#1130] [Q&A]
Hi all,
i have an Authoritative DNS Server for my Domains, works perfect, love it. But i want to use DynDns with my Unifi Dream Machine. I dont know how to get it work. i created a tsig key etc. i dont know which settings are correct in the Dream Machine. Does anybody have experience with this setup? Thanks!

**Comment by ShreyasZare** [comment]
Thanks for asking. If your router supports Dynamic Updated (RFC 2136) option then you can configure it to update your zone. Otherwise there is no direct option to do that. Usually, these routers support few commercial DDNS service providers by implementing their API.

The other option is to use the DNS server's API URL for [Add Record](https://github.com/TechnitiumSoftware/DnsServer/blob/master/APIDOCS.md#add-record) with the `ipAddress` parameter set to `request-ip-address` and configure a cron job or use task scheduler in windows to call that URL every couple of minutes. This will update the record with your outbound Internet IP address.

**Reply by ConGp20** [reply]
hi thanks for your answer. yes the router has support for dyndns updates. but i dont know how to fill in the settings:
<img width="394" alt="Bildschirmfoto 2024-11-29 um 14 50 29" src="https://github.com/user-attachments/assets/2e0ba09c-4621-4857-887a-cd8828170595">
i created a tsig key, set the update domain, set "allow" domain updates, in the zone settings in technitium dns. but now i dont now to fill in the settings on the router.

**Reply by ShreyasZare** [reply]
Check [this post](https://www.reddit.com/r/technitium/comments/1g0h5e2/acme_dns01_opnsense_to_technitium_help_guide/) on reddit which had similar issue.

You will need to check the documentation that is available from Unifi to know how to use those generic fields they have in the GUI.

**Reply by james-prince** [reply]
@ConGp20 did you ever get this working?
